addons/binding/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/protocol/RokuCommunication.java

```diff
@@ -0,0 +1,209 @@
+                            logger.error("Method not found {}", methodStringArray[ii].replace("-", "_"));
+                        }
+                    }
+                }
+            }
+        }
+        doc = getRequest(ROKU_ACTIVE_APP);
+        doc.getDocumentElement().normalize();
+        if (doc != null) {
+            NodeList nList = doc.getElementsByTagName("active-app");
+            for (int i = 0; i < nList.getLength(); i++) {
+                Node nNode = nList.item(i);
+                Element eElement = (Element) nNode;
+                logger.debug("Current Element: " + nNode.getNodeName());
+                try {
+                    state.active_app = getTagName("screensaver", eElement);
+                    String app_value = getSubTagName("screensaver", eElement);
+                    try {
+                        state.active_app_img = new RawType(getData("/query/icon/" + app_value));
+                    } catch (Exception e) {
+                        logger.debug("Failed to get channel artwork for: {}", e);
+                    }
+                } catch (NullPointerException e) {
+                    state.active_app = getTagName("app", eElement);
+                    String app_value = getSubTagName("app", eElement);
+                    try {
+                        state.active_app_img = new RawType(getData("/query/icon/" + app_value));
+                    } catch (Exception e1) {
+                        logger.debug("Failed to get channel artwork for: {}", e1);
+                    }
+                }
+            }
+        }
+        /**
+         * Future Functionality
+         * doc = getRequest(ROKU_QUERY_APPS);
+         * doc.getDocumentElement().normalize();
+         * if (doc != null) {
+         * NodeList nList = doc.getElementsByTagName("apps");
+         * for (int i = 0; i < nList.getLength(); i++) {
+         * Node nNode = nList.item(i);
+         * Element eElement = (Element) nNode;
+         * logger.debug("Current Element: " + nNode.getNodeName());
+         * HashMap<String, String> map = getAllSubTags("app", eElement);
+         * StringBuffer sb = new StringBuffer();
+         * for (String key : map.keySet()) {
+         * sb.append(key + " => " + map.get(key) + "\n");
+         * }
+         * state.application_menu = new StringType(sb.toString());
+         * }
+         * }
+         */
+    }
+
+    private Document getRequest(String context) throws IOException {
+        String response = processRequest(context);
+        logger.debug("HTTP Response: {}", response);
+        try {
+            DocumentBuilder db = dbf.newDocumentBuilder();
+            Document doc = db.parse(new InputSource(new StringReader(response)));
+            if (doc.getFirstChild().hasChildNodes() == false) {
+                throw new IOException("Could not handle response");
+            }
+            return doc;
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Please also note that you catch the exception thrown on line 168 in this catch...